mmaction/core/bbox/transforms.py

```diff
@@ -0,0 +1,119 @@
+import numpy as np
+import torch
+
+
+def bbox_overlaps(bboxes1, bboxes2, mode='iou', aligned=False):
+    """Calculate overlap between two set of bboxes.
+
+    Args:
+        bboxes1 (Tensor): shape (m, 4)
+        bboxes2 (Tensor): shape (n, 4), if 'aligned' is True, then m and n
+            must be equal.
+        mode (str): "iou" (intersection over union) or iof (intersection over
+            foreground).
+        aligned (bool): If 'aligned' is False, then calculate the ious between
+            each bbox of bboxes1 and bboxes2, otherwise the ious between each
+            aligned pair of bboxes1 and bboxes2.
+
+    Returns:
+        ious(Tensor): shape (m, n) if aligned == False else shape (m, 1)
+    """
+
+    assert mode in ['iou', 'iof']
+
+    rows = bboxes1.size(0)
+    cols = bboxes2.size(0)
+    if aligned:
+        assert rows == cols
+
+    if rows * cols == 0:
+        return bboxes1.new(rows, 1) if aligned else bboxes1.new(rows, cols)
+
+    if aligned:
+        lt = torch.max(bboxes1[:, :2], bboxes2[:, :2])  # [rows, 2]
+        rb = torch.min(bboxes1[:, 2:], bboxes2[:, 2:])  # [rows, 2]
+
+        wh = (rb - lt + 1).clamp(min=0)  # [rows, 2]
+        overlap = wh[:, 0] * wh[:, 1]
+        area1 = (bboxes1[:, 2] - bboxes1[:, 0] + 1) * (
+            bboxes1[:, 3] - bboxes1[:, 1] + 1)
```

Comment:
It seems in our open-mmlab repo, the area doesn't count the boundary (no `+1`) like `MMDetection`

Reply:
Not quite sure, but the code can pass the `test_bbox_overlaps` from mmcv. I read the unittest and find that it count the boundary. You can have a look at that [test_file](https://github.com/open-mmlab/mmcv/blob/761f725b70f4edb9ca7e9108dc8bcf7397093d8d/tests/test_ops/test_bbox.py)

Reply:
Now switch to the mmdet implementation